code/game/objects/structures/girders.dm

```diff
@@ -4,48 +4,55 @@
 	anchored = 1
 	density = 1
 	layer = 2
-	var/state = 0
+	var/state = 0 //state 0 = normal girder, state 1 = no idea, state 2 = reinforced girder, state 3 = displaced/dislodged girder
 	var/girderpasschance = 20 // percentage chance that a projectile passes through the girder.
 
 /obj/structure/girder/attackby(obj/item/W as obj, mob/user as mob, params)
 	add_fingerprint(user)
-	if(istype(W, /obj/item/weapon/wrench) && state == 0)
-		if(anchored && !istype(src,/obj/structure/girder/displaced))
-			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
-			user << "<span class='notice'>Now disassembling the girder...</span>"
-			if(do_after(user,40))
-				if(!src.loc)
-					return
-				user << "<span class='notice'>You dissasembled the girder!</span>"
-				new /obj/item/stack/sheet/metal(get_turf(src))
-				qdel(src)
-		else if(!anchored)
-			if (!istype(src.loc, /turf/simulated/floor))
-				usr << "<span class='danger'>A floor must be present to secure the girder!</span>"
-				return
-			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
-			user << "<span class='notice'>Now securing the girder...</span>"
+	if(istype(W, /obj/item/weapon/weldingtool) && state == 3)
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.remove_fuel(0,user))
+			playsound(src.loc, 'sound/items/Welder2.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] disassembles the girder.</span>", \
+								"You start to disassemble the girder...", "You hear welding and clanking.")
 			if(do_after(user, 40))
+				if( !WT.isOn() )
+					return
 				if(!src.loc)
 					return
-				user << "<span class='notice'>You secured the girder!</span>"
-				var/obj/structure/girder/G = new (loc)
-				transfer_fingerprints_to(G)
+				user << "<span class='notice'>You disassemble the girder.</span>"
+				var/obj/item/stack/sheet/metal/M = new (loc, 2)
+				transfer_fingerprints_to(M)
 				qdel(src)
 
+	else if(istype(W, /obj/item/weapon/wrench) && state == 3 && 4)

```

Comment:
Good eye! I forgot it there after testing if I could remove 90% of the cultgirder's mostly-duplicatecode and change cultgirder into just a state, but soon realized that this stuff was too advanced for me. I'll remove it.